src/provider/provider_coarse.c

```diff
@@ -1252,9 +1240,7 @@ static umf_result_t coarse_memory_provider_alloc(void *provider, size_t size,
             umf_result =
                 create_aligned_block(coarse_provider, size, alignment, &curr);
             if (umf_result != UMF_RESULT_SUCCESS) {
-                if (utils_mutex_unlock(&coarse_provider->lock) != 0) {
-                    LOG_ERR("unlocking the lock failed");
-                }
+                utils_mutex_unlock(&coarse_provider->lock);
```

Comment:
so there's no way this `unlock` ever fails...?

Reply:
It is very very very very very very very .... unlikely. If it happens, it will hang immediately on the lock() ...